siaskynet/_upload.py

```diff
@@ -47,17 +48,19 @@ def upload_request(self, upload_data, custom_opts=None):
         filename = opts['custom_dirname']
 
     params = {
-        'filename': filename,
         # 'skykeyname': opts['skykey_name'],
         # 'skykeyid': opts['skyket_id'],
     }
+    if filename:
+        params['filename'] = filename
 
     ftuples = []
     for filename, data in upload_data.items():
         ftuples.append((fieldname,
                         (filename, data)))
 
-    if len(upload_data) == 1:
+    if opts['portal_file_fieldname'] == fieldname:
```

Comment:
This isn't very elegant. I would change this check

Reply:
I've changed it.  The reason the code is awkward here is I wanted to keep it similar to the go code to ease new changes.  Let me know if you have more preferences around it.